online_meetings/canvas_zoom_meetings.py

```diff
@@ -43,95 +45,117 @@
-        for tab in tabs:
-            # Hidden only included if true
-            if (tab.label == "Zoom" and not hasattr(tab, "hidden")):
-                logger.info("Found a course with zoom as %s", tab.id)
-                r = CANVAS._Canvas__requester.request("GET", _url=tab.url)
-                external_url = r.json().get("url")
-                r = requests.get(external_url)
-                # Parse out the form from the response
-                soup = bs(r.text, 'html.parser')
-                # Get the form and parse out all of the inputs
-                form = soup.find('form')
-                if not form:
-                    logger.info("Could not find a form to launch this zoom page, skipping")
-                    break
-
-                fields = form.findAll('input')
-                formdata = dict((field.get('name'), field.get('value')) for field in fields)
-
-                # Get the URL to post back to
-                posturl = form.get('action')
-
-                # Start up the zoom session
-                zoom_session = requests.Session()
-                # Initiate the LTI launch to Zoom in a session
-                r = zoom_session.post(url=posturl, data=formdata)
-
-                # Get the XSRF Token
-                pattern = re.search('"X-XSRF-TOKEN".* value:"(.*)"', r.text)
-
-                zoom_courses.append({
-                    'account_id': course.account_id, 'course_id': course.id, 'course_name': course.name
-                })
-
-                if pattern:
-                    zoom_session.headers.update({
-                        'X-XSRF-TOKEN': pattern.group(1)
+    def __init__(self):
+        self.zoom_session = requests.Session()
+
+    def get_zoom_json(self, page_num: int = 1) -> Dict:
+        logger.info(f"Paging though course on page number {page_num}")
+        # Get tab 1 (Previous Meetings)
+        data = {'page': page_num,
+                'total': 0,
+                'storage_timezone': 'America/Montreal',
+                'client_timezone': 'America/Detroit'}
+        # TODO: Specify which page we want, currently hardcoded to previous meetings
+        r = self.zoom_session.get("https://applications.zoom.us/api/v1/lti/rich/meeting/history/COURSE/all", params=data)
+        # Load in the json and look for results
+        zoom_json = json.loads(r.text)
+        if zoom_json and "result" in zoom_json:
+            return zoom_json["result"]
+        return None
+
+    def get_zoom_details(self, url: str, data: Dict[str, str], course_id: int):
+        # Start up the zoom session
+        # Initiate the LTI launch to Zoom in a session
+        r = self.zoom_session.post(url=url, data=data)
+        # Get the XSRF Token
+        pattern = re.search('"X-XSRF-TOKEN".* value:"(.*)"', r.text)
+        if pattern:
+            self.zoom_session.headers.update({
+                'X-XSRF-TOKEN': pattern.group(1)
+            })
+
+            # Page iterator
+            page_num = 1
```

Comment:
I think this `page_num` can get deleted right, since you're using a loop var?

Reply:
Yup thanks that one wasn't used, thought I caught them all!